app/models/product_accessory.rb

```diff
@@ -9,10 +14,17 @@ class ProductAccessory < ActiveRecord::Base
   ## validations
   validates :product, presence: true
   validates :accessory, presence: true
-  validates :scaling_type, presence: true, inclusion: SCALING_TYPES
+  validates :scaling_type, presence: true
+  validate :scaling_type_matches_product
 
   def self.scaling_types
-    SCALING_TYPES
+    SCALING_TYPES.values.flatten.uniq
+  end
+
+  def scaling_type_matches_product
+    return if SCALING_TYPES[accessory.type.underscore].include?(scaling_type)
+
+    errors.add(:scaling_type)
```

Comment:
Hmm. I was kind of surprised by this since you didn't add any locales for it. But it turns out to be ok in terms of messaging:

```
> pa.errors.full_messages
=> ["Scaling Type is invalid"]
```